CombinedResourceHandler: MyFaces NPE

**MyFaces** 2.2.12
**OmniFaces** 2.7.12

I am on MyFaces 2.2.12 and unfortunately can't upgrade this project.  I am getting this intermittent NPE using OmniFaces 2.7.12.

```
Caused by: java.lang.NullPointerException
	at org.omnifaces.resourcehandler.CombinedResourceHandler.removeComponentResources(CombinedResourceHandler.java:428)
	at org.omnifaces.resourcehandler.CombinedResourceHandler.access$600(CombinedResourceHandler.java:230)
	at org.omnifaces.resourcehandler.CombinedResourceHandler$Builder.create(CombinedResourceHandler.java:661)
	at org.omnifaces.resourcehandler.CombinedResourceHandler$Builder.access$400(CombinedResourceHandler.java:558)
	at org.omnifaces.resourcehandler.CombinedResourceHandler$CombinedResourceBuilder.create(CombinedResourceHandler.java:544)
	at org.omnifaces.resourcehandler.CombinedResourceHandler.processEvent(CombinedResourceHandler.java:348)
	at javax.faces.event.SystemEvent.processListener(SystemEvent.java:40)
	at javax.faces.event.ComponentSystemEvent.processListener(ComponentSystemEvent.java:51)
	at org.apache.myfaces.application.ApplicationImpl._traverseListenerList(ApplicationImpl.java:2595)
	at org.apache.myfaces.application.ApplicationImpl.access$000(ApplicationImpl.java:129)
	at org.apache.myfaces.application.ApplicationImpl$SystemListenerEntry.publish(ApplicationImpl.java:2784)
	at org.apache.myfaces.application.ApplicationImpl.publishEvent(ApplicationImpl.java:610)
	at org.apache.myfaces.application.ApplicationImpl.publishEvent(ApplicationImpl.java:625)
	at org.jboss.as.jsf.injection.weld.ForwardingApplication.publishEvent(ForwardingApplication.java:299)
	at javax.faces.application.ApplicationWrapper.publishEvent(ApplicationWrapper.java:339)
	at org.apache.myfaces.lifecycle.RenderResponseExecutor.execute(RenderResponseExecutor.java:84)
	at org.apache.myfaces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:267)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:200)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at org.omnifaces.filter.FacesExceptionFilter.doFilter(FacesExceptionFilter.java:97)
```

I think the issue is this code...

```java
private static void removeComponentResources(FacesContext context, List<UIComponent> componentResourcesToRemove, String target) {
		UIViewRoot view = context.getViewRoot();

		for (UIComponent resourceToRemove : componentResourcesToRemove) {
			if (resourceToRemove != null) {
				UIComponent container = isMyFacesUsed() ? resourceToRemove.getParent() : resourceToRemove;

				// setInView(false) forces JSF to not save dynamic remove action in state.
				// Otherwise JSF will re-execute dynamic remove during restore view phase.
				// This is unnecessary as CombinedResourceHandler already takes care of it.
				// See also https://github.com/omnifaces/omnifaces/issues/135
				container.setInView(false);
				view.removeComponentResource(context, resourceToRemove, target);
				container.setInView(true);
			}
		}
	}
```

Its not doing a Null check if container is Null or doing a fallback of any kind for this line...

```java
UIComponent container = isMyFacesUsed() ? resourceToRemove.getParent() : resourceToRemove;
```

--- a/src/main/java/org/omnifaces/resourcehandler/CombinedResourceHandler.java
+++ b/src/main/java/org/omnifaces/resourcehandler/CombinedResourceHandler.java
@@ -425,9 +425,13 @@ public class CombinedResourceHandler extends DefaultResourceHandler implements S
 				// Otherwise JSF will re-execute dynamic remove during restore view phase.
 				// This is unnecessary as CombinedResourceHandler already takes care of it.
 				// See also https://github.com/omnifaces/omnifaces/issues/135
-				container.setInView(false);
+				if (container != null) {
+					container.setInView(false);
+				}
 				view.removeComponentResource(context, resourceToRemove, target);
-				container.setInView(true);
+				if (container != null) {
+					container.setInView(true);
+				}
 			}
 		}
 	}

--- a/src/main/java/org/omnifaces/resourcehandler/CombinedResourceHandler.java
+++ b/src/main/java/org/omnifaces/resourcehandler/CombinedResourceHandler.java
@@ -22,6 +22,7 @@ import static org.omnifaces.util.FacesLocal.isAjaxRequestWithPartialRendering;
 import static org.omnifaces.util.Hacks.isMyFacesUsed;
 import static org.omnifaces.util.Renderers.RENDERER_TYPE_CSS;
 import static org.omnifaces.util.Renderers.RENDERER_TYPE_JS;
+import static org.omnifaces.util.Utils.coalesce;
 import static org.omnifaces.util.Utils.isNumber;
 
 import java.util.ArrayList;
@@ -419,19 +420,15 @@ public class CombinedResourceHandler extends DefaultResourceHandler implements S
 
 		for (UIComponent resourceToRemove : componentResourcesToRemove) {
 			if (resourceToRemove != null) {
-				UIComponent container = isMyFacesUsed() ? resourceToRemove.getParent() : resourceToRemove;
+				UIComponent container = coalesce(isMyFacesUsed() ? resourceToRemove.getParent() : resourceToRemove);
 
 				// setInView(false) forces JSF to not save dynamic remove action in state.
 				// Otherwise JSF will re-execute dynamic remove during restore view phase.
 				// This is unnecessary as CombinedResourceHandler already takes care of it.
 				// See also https://github.com/omnifaces/omnifaces/issues/135
-				if (container != null) {
-					container.setInView(false);
-				}
+				container.setInView(false);
 				view.removeComponentResource(context, resourceToRemove, target);
-				if (container != null) {
-					container.setInView(true);
-				}
+				container.setInView(true);
 			}
 		}
 	}

